src/Lib.P2P/TaskHelper.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Lib.P2P
+{
+    /// <summary>
+    ///   Some helpers for tasks.
+    /// </summary>
+    public static class TaskHelper
+    {
+        /// <summary>
+        ///   Gets the first result from a set of tasks.
+        /// </summary>
+        /// <typeparam name="T">
+        ///   The result type of the <paramref name="tasks"/>.
+        /// </typeparam>
+        /// <param name="tasks">
+        ///   The tasks to perform.
+        /// </param>
+        /// <param name="cancel">
+        ///   Is used to stop the task.  When cancelled, the <see cref="TaskCanceledException"/> is raised.
+        /// </param>
+        /// <returns>
+        ///   A task that represents the asynchronous operation. The task's result is
+        ///   a <typeparamref name="T"/>>.
+        /// </returns>
+        /// <remarks>
+        ///   Returns the result of the first task that is not
+        ///   faulted or canceled.
+        /// </remarks>
+        public static async Task<T> WhenAnyResultAsync<T>(IEnumerable<Task<T>> tasks,
+            CancellationToken cancel)
+        {
+            var exceptions = new List<Exception>();
+            var running = tasks.ToList();
+            while (running.Count > 0)
+            {
+                cancel.ThrowIfCancellationRequested();
+                var winner = await Task.WhenAny(running).ConfigureAwait(false);
+                if (!winner.IsCanceled && !winner.IsFaulted) return await winner;
+                if (winner.IsFaulted)
+                {
+                    if (winner.Exception is AggregateException ae)
+                        exceptions.AddRange(ae.InnerExceptions);
+                    else
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'else' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)